sources/unstructured_data/inbox/__init__.py

```diff
@@ -0,0 +1,147 @@
+        client.select(folder, readonly=True)
+        criteria = [
+            f"(SINCE {start_date.strftime('%d-%b-%Y')})",
+            f"(UID {str(int(last_message_num))}:*)",
+        ]
+
+        if filter_emails:
+            logger.info(f"Load emails only from: {filter_emails}")
+            for email_ in filter_emails:
+                criteria.extend([f"(FROM {email_})"])
+
+        if gmail_group:
+            logger.info(f"Load all emails for Group: {gmail_group}")
+            criteria.extend([f"(TO {gmail_group})"])
+
+        status, messages = client.uid("search", *criteria)
+        message_uids = messages[0].split()
+
+        if not message_uids:
+            logger.warning("No emails found.")
+        else:
+            yield from [
+                {"message_uid": int(message_uid)} for message_uid in message_uids
+            ]
+
+
+@dlt.transformer(name="messages", write_disposition="replace")
+def read_messages(
+    item: TDataItem,
+    host: str = dlt.secrets.value,
+    email_account: str = dlt.secrets.value,
+    password: str = dlt.secrets.value,
+    body: bool = False,
+) -> TDataItem:
+    message_uid = str(item["message_uid"])
+
+    with imaplib.IMAP4_SSL(host) as client:
+        client.login(email_account, password)
+        msg = get_message_obj(client, message_uid)
+        if msg:
+            result = deepcopy(item)
+            result.update(extract_email_info(msg, body=body))
+            result.update({"internal_date": get_internal_data(client, message_uid)})
+            yield result
+
+
+@dlt.transformer(name="attachments", write_disposition="replace")
+def get_attachments_by_uid(
+    item: TDataItem,
+    storage_folder_path: str,
+    host: str = dlt.secrets.value,
+    email_account: str = dlt.secrets.value,
+    password: str = dlt.secrets.value,
+    body: bool = False,
+) -> TDataItem:
+    message_uid = str(item["message_uid"])
+
+    with imaplib.IMAP4_SSL(host) as client:
+        client.login(email_account, password)
+        msg = get_message_obj(client, message_uid)
+        if msg:
+            email_info = extract_email_info(msg, body=body)
+            internal_date = get_internal_data(client, message_uid)
+
+            for part in msg.walk():
+                content_disposition = part.get("Content-Disposition", "")
+                if "attachment" in content_disposition:
+                    filename = part.get_filename()
+                    if filename:
+                        attachment_data = part.get_payload(decode=True)
+                        attachment_path = os.path.join(
+                            storage_folder_path, message_uid + filename
+                        )
+                        os.makedirs(os.path.dirname(attachment_path), exist_ok=True)
+
+                        with open(attachment_path, "wb") as f:
+                            f.write(attachment_data)
+
+                        result = deepcopy(item)
+                        result.update(email_info)
```

Comment:
maybe you could put it into `envelope` field not to mix attachment data with message (envelope in this context)

Reply:
only attachment part? or whole info (don't think it makes sense)?